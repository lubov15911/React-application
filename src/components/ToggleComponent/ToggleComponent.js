import React from 'react';
import PropTypes from 'prop-types';

import './ToggleComponent.scss';

const ToggleComponent = (props) => {
    const {
        selected,
        toggleType,
        options,
        handleToggle,
    } = props;

    function isChecked(option) {
        return selected === option;
    }

    return (
        <div className="toggler">
            <p>
                {toggleType}
                y
            </p>
            <input
                type="radio"
                id={`first-option-${toggleType}`}
                name={`toggler-${toggleType}`}
                value={options.first}
                checked={isChecked(options.first)}
                onChange={handleToggle}
            />
            <label htmlFor={`first-option-${toggleType}`}>{options.first}</label>
            <input
                type="radio"
                id={`second-option-${toggleType}`}
                name={`toggler-${toggleType}`}
                value={options.second}
                checked={isChecked(options.second)}
                onChange={handleToggle}
            />
            <label htmlFor={`second-option-${toggleType}`}>{options.second}</label>
        </div>
    );
};
ToggleComponent.propTypes = {
    selected: PropTypes.string.isRequired,
    toggleType: PropTypes.string.isRequired,
    options: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
    handleToggle: PropTypes.func.isRequired,
};

export default ToggleComponent;
