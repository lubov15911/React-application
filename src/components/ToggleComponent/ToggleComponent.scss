@import "../../_colors";

.toggler {
  display: flex;
  font-size: 1.5em;
  line-height: 2.5em;
  text-transform: uppercase;
  margin: 1em 0;
  opacity: 0.8;

  input {
    display: none;
  }

  p {
    margin-right: 1em;
  }

  label {
    border-radius: 0.4em 0 0 0.4em;
    background-color: rgba(66, 66, 66, 0.68);
    padding: 0 1em;
    cursor: pointer;
  }

  label:last-child {
    border-radius: 0 0.4em 0.4em 0;
  }

  input:checked + label {
    background-color: $pink;
  }
}
