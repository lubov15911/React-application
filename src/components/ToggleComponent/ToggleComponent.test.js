import React from 'react';
import { shallow } from 'enzyme';

import ToggleComponent from './ToggleComponent';

describe('SearchHeader', () => {
    it('should render correctly', () => {
        const toggleType = 'Sort';
        const SortOptions = {
            first: 'Release date',
            second: 'Rating'
        };
        const fakeFunction = () => {};

        const component = shallow(<ToggleComponent
            toggleType='Sort'
            options={SortOptions}
            selected={SortOptions.first}
            handleToggle={fakeFunction}
        />);

        expect(component).toMatchSnapshot();
    });
});
