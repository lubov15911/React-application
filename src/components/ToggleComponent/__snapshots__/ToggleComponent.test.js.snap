// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader should render correctly with no props 1`] = `
<div
  className="toggler"
>
  <p>
    Sort
    By
  </p>
  <input
    checked={true}
    id="first-option-Sort"
    name="toggler-Sort"
    onChange={[Function]}
    type="radio"
    value="Release date"
  />
  <label
    htmlFor="first-option-Sort"
  >
    Release date
  </label>
  <input
    checked={false}
    id="second-option-Sort"
    name="toggler-Sort"
    onChange={[Function]}
    type="radio"
    value="Rating"
  />
  <label
    htmlFor="second-option-Sort"
  >
    Rating
  </label>
</div>
`;
